mcx/distributions/constraints.py

```diff
@@ -175,14 +177,52 @@ def __call__(self, x):
         return np.all(x > 0, axis=-1) & (x_sum <= 1) & (x_sum > 1 - 1e-6)
 
 
+class _IndependentConstraint(Constraint):
+    def __str__(self):
+        return "an event is valid only if all its independent entries are valid."
+
+    def __init__(self, base_constraint, ndims):
+        assert isinstance(base_constraint, Constraint)
+        assert isinstance(ndims, int)
+        assert ndims >= 0
+
+        self.base_constraint = base_constraint
+        self.ndims = ndims
+        super().__init__()
+
+    def __call__(self, value):
+        result = self.base_constraint(value)
+        if self.ndims == 0:
+            return result
+        elif np.ndim(result) < self.ndims:
+            raise ValueError(
+                (f"Expected value.dim() >= {self.ndims}" " but got {np.ndims{result}}")
+            )
+        result = result.all(-1)
+        return result
+
+
+class _PositiveDefinite(Constraint):
+    event_dim = 2
+
+    def __call__(self, x):
+        jnp = numpy if isinstance(x, (numpy.ndarray, numpy.generic)) else np
```

Comment:
Does it otherwise fail if you pass a numpy array?

Reply:
This `jnp` / `np` mess reminds me that I need to switch to `import jax.numpy as jnp` everywhere :)

Reply:
I am basically ripping off the code from [`numpyro`](https://github.com/pyro-ppl/numpyro/blob/master/numpyro/distributions/constraints.py). This is how they define constraints there. Should I test what happens without this line? Actually, should we have test cases for constraints?

Reply:
Also, want me to create an issue for the `jnp` business?

Reply:
Yes if you could check what happens without this line it would be great. No need to add a test for constraints here, just making sure it works once should be enough!

And yes opening an issue marking it as "good first issue" would be great 😊

Reply:
Your intuition was right. Works without that line and just using jax.numpy for all.
Will clean it up.

Reply:
I created issue #72 but don't have permissions to add labels etc. You might have to do that.